packit/utils/changelog_helper.py

```diff
@@ -58,7 +60,12 @@ def update_dist_git(self, full_version: str, upstream_tag: str) -> None:
             )
         )
         try:
-            self.dg.set_specfile_content(self.up.specfile, full_version, comment)
+            self.dg.set_specfile_content(
+                self.up.specfile,
+                full_version,
+                comment,
```

Comment:
Would it work if you'd set `comment` to `None` here if `up.specfile.has_autochangelog()` is True, and avoid introducing the dedicated flag?

This way the logic would be as: "add no new comment to the %changelog if the upstream spec file uses %autochangelog", and you could drop all the changes from `set_specfile_content`.

Reply:
I think it will not work.
I need to replace the content of the changelog section with the autochangelog macro only if it exists in the upstream specfile.
set_specfile_content can be called with no comment to just update the version and, in this case, should not change the changelog section to %autochangelog.